lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -600,6 +600,25 @@ SILCombiner::optimizeConcatenationOfStringLiterals(ApplyInst *AI) {
   return tryToConcatenateStrings(AI, Builder);
 }
 
+/// Determine the pattern for global_addr..
+/// %3 = global_addr @$P : $*SomeP
+/// %4 = init_existential_addr %3 : $*SomeP, $SomeC
+/// %5 = alloc_ref $SomeC
+/// store %5 to %4 : $*SomeC
+static SILValue findInitExistentialFromGlobalAddr(GlobalAddrInst *GAI) {
+  /// Check for a single InitExistential.
+  InitExistentialAddrInst *SingleIE = nullptr;
+  for (auto Use : GAI->getUses()) {
+    auto *User = Use->getUser();
+    if (auto *InitExistential = dyn_cast<InitExistentialAddrInst>(User)) {
```

Comment:
Sorry, I missed this point earlier. You are right that in some obfuscation the global_addr, that is the source of copy_addr, could be used in an init_existential that occurs after the copy_addr instruction.  In that case, we can either make sure both global_addr and init_existential dominate the copy_addr (heavy weight) or restrict all of them to our desired order within a basic block (light-weight). Since this is a peephole, I suggest we use the later approach. Thoughts?